CHANGES.md

```diff
@@ -1,18 +1,29 @@
 ringpop-go changes
 ==================
 
-v0.DEV (to be released)
------------------------
+v0.4.0
+------
 
-* Remove `File` and `Host`-based discover providers in favor of
-  `DiscoverProvider` interface #119
-* Ringpop will always assume the current host (`ringpop.node.identity`) is part
-  of the cluster. Previously, it was true for only `Host`-based discovery.
-* Add Go 1.6 testing on CI
+* Feature: Faulty nodes are now automatically reaped from nodes' membership
+  lists after 24 hours (by default). #123
+* New options for controlling suspect and reaping times. #123
+* Add new `Ready` and `Destroyed` events to ringpop. #125
+* Add additional logging to bring ringpop-go on par with ringpop-node log
+  messages. #116
+* Fix bug where Ringpop automatically added itself to the bootstrap hosts for
+  host-based bootstrapping, but not other bootstrapping methods. #120
+* Fix race condition where membership and hashring could be inconsistent with
+  each other. #112
+* Remove `File` and `Host` options from bootstrap options in favor of
+  `DiscoverProvider` interface. #120
+* Add Go 1.6 to testing on CI
```

Comment:
Yes, that's what I mean. I'll try to be the devil's advocate here (a.k.a. customer's). Changelog should help me to answer the question: _how should I change my code base to stay compatible?_ Let's go 1-by-1:

> Feature: Faulty nodes are now automatically reaped from nodes' membership lists after 24 hours (by default). #123

... This is changing behavior, not reflected in release notes (e.g. users might need to change their tooling).

> New options for controlling suspect and reaping times.

Nice! As long as defaults didn't change, it's great.

> Add new `Ready` and `Destroyed` events to ringpop. #125

Another sentence on what you can do with them. E.g. "these are optional events that can invoke a user callback when ringpop is ready and destroyed, respectively". Otherwise, it's a compatible change.

> Fix bug where Ringpop automatically added itself to the bootstrap hosts for

Nice, bug fix! But wait if I relied on `self` being added, what do I do now? Should I remove `self`? What if I had the DiscoverProvider in 0.3, how should I change in 0.4? I know I implemented that feature, and could have documented it the way I see it.

The remaining changelog looks perfect.